Add SE2 projects stats below intro video

Pabl0cks opened the pull request:
Adds a new stats component showing Scaffold-ETH 2 repositories metrics (projects built, GitHub stars, forks) fetched from the se2-projects API.

Changes:
- New SE2Stats component displaying repositories metrics
- Stats are fetched via getStaticProps (revalidates every 6 hours)
- Positioned below the intro video for better content flow
- Links to se2-projects.vercel.app for full project stats + repository list

Also reduced the spacing before the video so it shows a bit above the fold on big screens, trying to get more views or scrolls down. Feel free to undo!

Pabl0cks committed:
Add SE2 stats component and show at homepage

carletex commented:
Great stuff! Wasn't aware of this change (but I like it)

It could even go before the video.

Let's also have a good subdomain for https://se2-projects.vercel.app/ (projects.scaffoldeth.io?)

Pabl0cks commented:
> Let's also have a good subdomain for https://se2-projects.vercel.app/ (projects.scaffoldeth.io?)

Yeah, I like the subdomain! 🙏 Wasn't so confident about linking to the site or keeping it internal just for the stats, but I think is nice to have in terms of transparency and more data granularity.

damianmarti commented:
Looking great!

> > Let's also have a good subdomain for https://se2-projects.vercel.app/ (projects.scaffoldeth.io?)
> 
> Yeah, I like the subdomain! 🙏 Wasn't so confident about linking to the site or keeping it internal just for the stats, but I think is nice to have in terms of transparency and more data granularity.

This is deployed on my Vercel account now. I can deploy it to BG account, and we can use the subdomain projects.scaffoldeth.io

technophile-04 commented on packages/nextjs/components/SE2Stats.tsx:
```diff
@@ -0,0 +1,69 @@
+export const SE2Stats = ({ stats }: { stats: SE2StatsData | null }) => {
+  return (
+    <div className="w-full max-w-xl mx-auto pt-10">
+      <p className="text-lg font-light text-center mb-3 tracking-wide">Join thousands of builders</p>
```
Nitpick, how about: 

"Trusted By Ethereum community" or "Builders are shipping apps with Scaffold-ETH 2"?

technophile-04 commented:
> It could even go before the video. 

++ we shall def keep it above video

--- 

Thanks @damianmarti and @Pabl0cks for the great work on this! Love it!

Pabl0cks committed:
Move above video, tweak copy

Pabl0cks commented:
Thanks for the feedback! I've just pushed a couple of changes:

- Moved the component above the video. Also moved "Explore all the extensions" to the right so they don't "compete", that was the original reason of placing it below the video, didn't love how both texts looked one above each other
- Tweaked the copy to "Trusted by a big community of builders". Open to do a final tweak 🙌
- Updated the link and api call to the final url: https://projects.scaffoldeth.io/

<img width="1849" height="875" alt="image" src="https://github.com/user-attachments/assets/217cb7a1-17f3-49ff-a267-e768a01857f8" />

carletex reviewed (approved):
lgtm!

technophile-04 reviewed (approved):
Merging this 🙌! Damu Already move the repo to https://github.com/scaffold-eth/se2-projects
